tests/forms/uploadfield/UploadFieldTest.php

```diff
@@ -167,7 +167,9 @@ public function testAllowedExtensions() {
 			'UploadFieldTest_Controller/Form/field/AllowedExtensionsField/upload',
 			array('AllowedExtensionsField' => $this->getUploadFile($invalidFile))
 		);
-		$this->assertTrue($response->isError());
+		$responseJson = Convert::json2array($response->getBody());
+		
+		$this->assertTrue($response->isError() || isset($responseJson[0]['error']));
```

Comment:
I think this kind of assertion is not a good idea. Putting OR logic in here means that we never have any guarantee that the last condition is ever true because as soon as the first one is the second is ignored.

In a unit test you should know the expected outcome and test that, there shouldn't be any chance about this assertion.

Therefore, either both should be individually asserted or the one that doesn't happen should be removed (and perhaps assertFalse instead)